sway-core/src/ir_generation/function.rs

```diff
@@ -390,29 +435,34 @@ impl FnCompiler {
         let lhs_val = self.compile_expression(context, ast_lhs)?;
         let rhs_block = self.function.create_block(context, None);
         let final_block = self.function.create_block(context, None);
-        let cond_builder = self.current_block.ins(context);
-        match ast_op {
-            LazyOp::And => cond_builder.conditional_branch(
-                lhs_val,
-                rhs_block,
-                final_block,
-                Some(lhs_val),
-                span_md_idx,
-            ),
-            LazyOp::Or => cond_builder.conditional_branch(
-                lhs_val,
-                final_block,
-                rhs_block,
-                Some(lhs_val),
-                span_md_idx,
-            ),
-        };
+        if !self.current_block.is_terminated_by_a_branch(context) {
```

Comment:
This needs to be done to avoid a double branch due to a break/continue, yeah?  And the verifier would complain in that case.  It _does_ feel like this sort of thing would be better handled by DCE... I'm starting to wonder if the block verifier is too strict... maybe we only need to call `verify` once, after we've finished opts and are ready to hand over to ASMgen... thoughts?

Reply:
I feel like IRgen shouldn't have to stress out over introducing dead code.

Reply:
My reasoning here is that a basic block (at least in LLVM) must have a single terminator. A block with multiple `br` instructions in it is malformed by definition. We can choose to change that if we like.

On the topic of only running the verifier at the end... I actually think we should go in the opposite direction and run the verifier as many times as possible to make sure that no optimization pass (nor IRgen itself) is generating illegal IR.

Reply:
OK, fair call. 🙂 I initially thought the verifier should be run during optimisation by the pass manager, when we get one.   

OTOH seeing friction in IRgen is annoying.  We _could_ eventually allow crap from IRgen, not verify prior to opts, but mandate DCE at the top of the pass manager list and then verify like crazy.  That seems a bit convoluted, but if it means we can have dead terminators and empty blocks, etc. come out of IRgen without breaking anything it might be easier overall.